modules/table/src/main/java/org/apache/ignite/internal/table/distributed/disaster/GroupUpdateRequest.java

```diff
@@ -143,363 +102,48 @@ public boolean manualUpdate() {
-            // TODO: IGNITE-23635 Add fail handling for failed resetPeers
-            if (throwable != null) {
-                LOG.error("Failed to reset partition", throwable);
-            }
-        });
-    }
-
     /**
-     * Sets force assignments for the zone/table if it's required. The condition for force reassignment is the absence of stable
-     * assignments' majority within the set of currently alive nodes. In this case we calculate new assignments that include all alive
-     * stable nodes, and try to save ot with a {@link Assignments#force()} flag enabled.
-     *
-     * @param tableId Table id.
-     * @param zoneDescriptor Zone descriptor.
-     * @param dataNodes Current DZ data nodes.
-     * @param aliveNodesConsistentIds Set of alive nodes according to logical topology.
-     * @param revision Meta-storage revision to be associated with reassignment.
-     * @param timestamp Meta-storage timestamp to be associated with reassignment.
-     * @param metaStorageManager Meta-storage manager.
-     * @param localStatesMap Local partition states retrieved by
-     *         {@link DisasterRecoveryManager#localTablePartitionStates(Set, Set, Set)}.
-     * @return A future that will be completed when reassignments data is written into a meta-storage, if that's required.
+     * Returns {@code true} if this request is a zone request with enabled colocation and {@code false} if this is a table request with
+     * colocation disabled.
      */
-    private static CompletableFuture<Void> forceAssignmentsUpdate(
-            int tableId,
-            CatalogZoneDescriptor zoneDescriptor,
-            Set<String> dataNodes,
-            Set<String> aliveNodesConsistentIds,
-            long revision,
-            HybridTimestamp timestamp,
-            MetaStorageManager metaStorageManager,
-            Map<TablePartitionId, LocalPartitionStateMessageByNode> localStatesMap,
-            long assignmentsTimestamp,
-            int[] partitionIds,
-            boolean manualUpdate
-    ) {
-        return tableStableAssignments(metaStorageManager, tableId, partitionIds)
-                .thenCompose(tableAssignments -> {
-                    if (tableAssignments.isEmpty()) {
-                        return nullCompletedFuture();
-                    }
-
-                    return updateAssignments(
-                            tableId,
-                            zoneDescriptor,
-                            dataNodes,
-                            aliveNodesConsistentIds,
-                            revision,
-                            timestamp,
-                            metaStorageManager,
-                            localStatesMap,
-                            assignmentsTimestamp,
-                            partitionIds,
-                            tableAssignments,
-                            manualUpdate
-                    );
-                });
+    boolean colocationEnabled() {
```

Comment:
This looks like a breaking change from AI 3.0. How are we going to fix this to make sure 3.1 is backwards compatible?

Maybe we should make the colocation flag explicit and add new serialization format version. This is some additional code, but the flag will be explicit and no backwards compatibility will be immediately broken